### ISSUE TITLE
Glean error in search.in_content metric

### ISSUE THREAD

**Issue by travis79** [mozilla-mobile/firefox-ios#8724]
While looking at [telemetry error metrics for Firefox iOS](https://sql.telemetry.mozilla.org/queries/80603/source#199908) I noticed a high error rate for the `Search.inContent` labeled counter metric.  Based on the error type, it looks like there is an invalid "key" being used. Glean checks the value passed into the dictionary type [against a regex](https://mozilla.github.io/glean/book/reference/metrics/index.html?highlight=labels#label-format) that limits the characters and length. 

I believe that the problem either is here on this line:
https://github.com/mozilla-mobile/firefox-ios/blob/373948df8c2fa4e52f6b327fd30527e6a1f2d785/Client/Frontend/Browser/SearchTelemetry/SearchTelemetry.swift#L42

or on this line:
https://github.com/mozilla-mobile/firefox-ios/blob/373948df8c2fa4e52f6b327fd30527e6a1f2d785/Client/Frontend/Browser/SearchTelemetry/SearchTelemetry.swift#L47

but could possibly be on this line also, if the `provider` causes it to exceed the length limitations:
https://github.com/mozilla-mobile/firefox-ios/blob/373948df8c2fa4e52f6b327fd30527e6a1f2d785/Client/Frontend/Browser/SearchTelemetry/SearchTelemetry.swift#L52

The error is probably due to either an invalid character being inserted into the "key/label" or, more likely, by exceeding the length restriction.

Just to be clear, the telemetry is still being recorded under a generic `__other__` label, so nothing is currently being lost, but it may be getting inaccurately mixed together under the generic label. To ensure that this is being recorded correctly the labels used should match the requirements, or if the requirements demand it, file an issue with Glean to increase the limitations.

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FXIOS-1990)

**Comment by athomasmoz** [comment]
Note: this means that different types of in_context search counts could be off. For example, some organic searches may be incorrectly tagged as other

**Comment by data-sync-user** [comment]
➤ Nishant Bhasin commented:

The labels that are being sent should follow the following regex from Glean. You can learn more about it in the link below.

^[a-z_][a-z0-9_-]{0,29}(\\.[a-z_][a-z0-9_-]{0,29})*$

https://mozilla.github.io/glean/book/reference/metrics/index.html#label-format ( https://mozilla.github.io/glean/book/reference/metrics/index.html#label-format|smart-link )

**Comment by badboy** [comment]
This is persisting still.
I looked at the code and think I found the mistake here: https://github.com/mozilla-mobile/firefox-ios/blob/373948df8c2fa4e52f6b327fd30527e6a1f2d785/Client/Frontend/Browser/SearchTelemetry/SearchTelemetry.swift#L20-L21

In case of an unknown search engine (`SearchEngine.none`) the `code is set to the empty string, resulting in a label that ends in a dot (`.`). That's an invalid label.
It should be as easy as using `"none"` as the code.
